src/components/buttons.tsx

```diff
@@ -19,6 +19,41 @@ enum ButtonVariant {
   GHOST_COOKIE = "GHOST_COOKIE",
 }
 
+function getButtonStyle(variant?: ButtonVariant): string {
```

Comment:
Still not in favor of `@apply` @Fenntasy ?
Because here, I think it could make sense to make something like that: https://github.com/Meyclem/where-is-it/blob/main/src/styles/globals.css#L65

Reply:
One thing I can do to is you guys really dislike this, is to split them into their own components.

Reply:
> It's just that writing style in function while we could have written it in css make me wonder if we take the right direction.

Still not in favor of calling `@apply` CSS in fact. I agree with you that doing complicated things with JS will result in an organisation mess but I don't agree your `.btn` class isn't already another organisational mess.

I really feel the goal of tailwind is to lay bare the classes to be seen directly on the relevant element. Abstracting it away with `@apply` is a door into having something like:

```
<button className="btn shadow border-none">
```
Where we will fight against the abstracted class to change it instead of editing it where it is which would result in a lot of mess as well.

But, it's not something we can really avoid there, we could either have one `Button` component that is extended by the others or have them all be based on `button` and repeat some classes. Either way, we need to have a mess because we're doing inheritance there.

We can do it in CSS, with functions or with OOP (`BaseButton` component) but this is the root of the mess imho (I can always be wrong but more on this specific case, take this opinion with the required amount of salt).

My preference would go on having them all be buttons because I'd prefer to not have `rounded rounded-t-none` but I'll defer on your preferences.

However, I think another thing that is annoying is the presence of primary-cookie, having a style there for one instance of a button seems overkill to me. What if we end up with other buttons that are "primary-cookie" that don't have anything to do with the cookie part?

Reply:
I agree with you on there.
The thing is we will have to lay down some rules either way to "limit" the possibility to mess things up.